tools/cldr-apps/src/main/java/org/unicode/cldr/web/AnnouncementData.java

```diff
@@ -0,0 +1,465 @@
+        }
+    }
+
+    private static int savePostToDb(
+            Announcements.AnnouncementSubmissionRequest request, UserRegistry.User user)
+            throws SurveyException {
+        int announcementId;
+        try {
+            Connection conn = null;
+            PreparedStatement pAdd = null;
+            try {
+                conn = CookieSession.sm.dbUtils.getAConnection();
+                pAdd = prepare_pAdd(conn);
+                pAdd.setInt(1, user.id); // "poster"
+                DBUtils.setStringUTF8(pAdd, 2, request.subject); // "subj"
+                DBUtils.setStringUTF8(pAdd, 3, request.body); // "text"
+                pAdd.setString(4, request.locales); // "loc"
+                pAdd.setBoolean(5, request.orgsAll); // "orgsAll"
+                pAdd.setString(6, request.audience); // "audience"
+                int n = pAdd.executeUpdate();
+                if (n != 1) {
+                    throw new RuntimeException("Couldn't post announcement, update failed.");
+                }
+                announcementId = DBUtils.getLastId(pAdd);
+                emailNotify(request, user, announcementId);
+            } finally {
+                DBUtils.close(pAdd, conn);
+            }
+        } catch (SQLException se) {
+            String complaint =
+                    "Couldn't save announcement post to db - " + DBUtils.unchainSqlException(se);
+            SurveyLog.logException(logger, se, complaint);
+            throw new SurveyException(SurveyException.ErrorCode.E_INTERNAL, complaint);
+        }
+        return announcementId;
+    }
+
+    private static PreparedStatement prepare_pAdd(Connection conn) throws SQLException {
+        String sql =
+                "INSERT INTO "
+                        + DBUtils.Table.ANNOUNCE
+                        + " (poster,subj,text,loc,orgsAll,audience)"
+                        + " values (?,?,?,?,?,?)";
+        return DBUtils.prepareStatement(conn, "pAddAnnouncement", sql);
+    }
+
+    private static void emailNotify(
+            Announcements.AnnouncementSubmissionRequest request,
+            UserRegistry.User poster,
+            int announcementId) {
+        ElapsedTimer et = new ElapsedTimer("Sending email for announcement " + announcementId);
+        Set<Integer> recipients = new HashSet<>();
+        gatherRecipients(request, poster, recipients);
+        logger.fine(
+                et
+                        + ": Announcement notify: u#"
+                        + poster.id
+                        + " announcement:"
+                        + announcementId
+                        + " queueing:"
+                        + recipients.size());
+        if (recipients.size() == 0) {
+            return;
+        }
+        String subject = "CLDR announcement: " + request.subject;
+        String body =
+                "There is a new announcement in the Survey Tool. Do not reply to this message, instead go to the Survey Tool";
```

Comment:
Note: the email does not contain the text of the announcement -- should it?

If we do want the email to contain the notification text (not only the subject line), then we need to make sure that any html mark-up in the notification is also displayed correctly in the email

Reply:
Per yesterday's meeting, the body will be plain text, and it will be included in the email notification

Reply:
that's fixed in my subsequent commit